### DISCUSSION TITLE
Unable to rate-limit on arg/kwarg value

### DISCUSSION THREAD

**Discussion by schinckel** [jsocol/django-ratelimit#281] [Q&A] [answered]
Is it possible to rate-limit on the value of an arg/kwarg coming from the URL router?

For instance, I have a view `send_sms_notification` that has `user_id` as an arg/kwarg, rather than in the POST keys.

    def send_sms_notification(request, user_id):
        if user.can_send_sms_to(user_id):
            send_sms_to(user_id, request.POST.get('data'))

(That's not real code, just an analog to what I have).

The `user_id` in that context is not the logged in user, but another user. It's not available in the request.POST. Is it possible to use these type of values as keys?

**Comment by schinckel** [comment, marked as answer]
Aha! When using a callable, the second argument is the view object (at least, in the case of my CBV), and we can reference that.

**Comment by benjaoming** [comment]
Congrats! I guess the issue can be closed :)

For future reference, you can ask questions here: https://github.com/jsocol/django-ratelimit/discussions
